pkg/in_memory_state_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2025 Red Hat Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcd_shield_test
+
+import (
+	"context"
+
+	etcd_shield "github.com/konflux-ci/etcd-shield/pkg"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"pgregory.net/rapid"
+)
+
+var _ = Describe("in-memory smoke tests", func() {
+	var state etcd_shield.StateManager
+
+	BeforeEach(func() {
+		state = etcd_shield.NewInMemoryState()
+	})
+
+	It("should load saved state", func(ctx context.Context) {
+		Expect(state.WriteConfig(ctx, false)).To(Succeed())
+		Expect(state.ReadConfig(ctx)).To(Equal(false))
+	})
+
+	It("should never return an error", func(ctx context.Context) {
+		rapid.Check(GinkgoT(), func(t *rapid.T) {
```

Comment:
Nice, what's that doing?

Reply:
This generates a random sampling of bools a bunch of times (default is 100, but it's tunable), and ensures the test you pass in doesn't fail by panicking or calling methods such as `t.Fatalf`.  This is property testing, and the property I want to test here is that we don't return errors when using this manager.

On consideration, it's probably redundant to have this distinct from the prior test.